### PR TITLE
meta-modules in node

### PR DESCRIPTION
**Pull request by tech4GT**
@jywarren The approach i have taken for this one is having a meta-modules.json file and when user gives the input the meta-modules are expanded in program.step without destroying any ordering then normal operation occurs
I have defined one named `example` which equals `green-channel invert`
Let's say program.step initially has value [`brightness`, `example`, `blur`] then expandMetaModules is called, then program.step will become [`brightness`,`green-channel`,`invert`,`blur`]
Is this a good enough approach for this?, please review
Signed-off-by: tech4GT <varun.gupta1798@gmail.com>

### CODE REVIEW

**Comment by tech4GT** [comment]
#203

**Comment by tech4GT** [comment]
@jywarren also a perk to having a file to store these modules is that if one user wishes to save or his/her configuration( like we do with IDEs like webstorm and VSCode ) they can simply share this file.

File: src/config/MetaModules.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[
+    {
+        "name": "example",
+        "description": "An example meta module",
+        "steps":[
```
Comment (jywarren) [inline src/config/MetaModules.json]: So while this is cool, I wonder if we can specify meta modules in a serialized string that is readable as part of a url request. Maybe we need both options so like this OR `?steps=invert[settings...],green-channel, threshold [50]` just for example.

**Comment by tech4GT** [comment]
@jywarren presets are given just as we give for a sequence of steps since all this is doing is making an alias for a sequence.

File: src/config/MetaModules.json (hunk 1 above)
Comment (tech4GT) [reply src/config/MetaModules.json]: @jywarren you are talking about the browser, right? Actually we have a separate issue to do this for the browser and it's a part of summer proposal as well, if you say i can start working on this right now?

**Comment by tech4GT** [comment]
@jywarren I was thinking maybe we can have an option to store default value of options in the json itself and if the user does not give any values when the module is used it can be used as sort of a fallback case or default values for the meta-module

File: src/config/MetaModules.json (hunk 1 above)
Comment (jywarren) [reply src/config/MetaModules.json]: Actually, the ability to record both steps and settings in a string could be useful for browser work, for saving sequences in `localStorage`, in a key/value database on server or client side, and for REST based processing, so I think it's worth considering for all those reasons.

Great, though -- so for this PR, so as not to create an incompatibility, we should either think about a subset of the full syntax that will remain compatible after the settings-in-a-string feature, so while here we might just enable settings steps in JSON, we should expect to be able to detect a string submission and switch into a different parsing mode which can read in the settings. I think that means we are OK here - but maybe we should leave an inline comment saying we'll soon accept compact string format like `?steps=invert(settings...),green-channel,threshold(50)` 

Perhaps using parentheses will make it more like a function call... but we don't have to make a final decision just yet.

Comment (jywarren) [reply src/config/MetaModules.json]: See #127 for current planning on compact settings options in URLs

**Comment by jywarren** [comment]
^ great idea -- adding to #127!!

**Comment by tech4GT** [comment]
@jywarren this sounds great, I'll add the comment, I had some questions though
- Saving to local storage is the browser counterpart of saving to the json file right?
- do we want string based steps in node environment as well, what I was thinking was that node and browser meta-modules implementation can be separate, on the node side we store them with default settings in the json file as json and on the browser side we store them as parable strings in the local storage, is this right, sorry I wasn't able to understand completely?

**Comment by jywarren** [comment]
Sorry this took me a while!!!

**Comment by tech4GT** [comment]
@jywarren no prob, working on this now I'll try to come up with a meta module checklist in this issue itself so we can build up from here✌️
Thanks

**Comment by tech4GT** [comment]
@jywarren should I put these in an improvents for summer issue ? since this overlaps with some stuff there, I was planning on makiuthat issue as well as you suggested

**Comment by jywarren** [comment]
Yes, i think most of this could be done in follow-up, and we could start with the most basic meta-module implementation and create a checklist for what to do after merging this. How does that sound?

**Comment by tech4GT** [comment]
That sounds great 😁

**Comment by jywarren** [comment]
🚀

**Comment by tech4GT** [comment]
@jywarren actually i cannot comment on the .json file😅

**Comment by jywarren** [comment]
Let's just open up a new issue exactly describing this needed upgrade, or clean up the one where we spec out the string-based system. OK?

**Comment by tech4GT** [comment]
@jywarren i have kind of made a checklist related to this here #200 once you approve we'll edit the issue and add the ckecklist there, i think it pretty much consolidates all of our ideas😁

**Comment by jywarren** [comment]
Awesome, I replied over there. Do you want to merge this in, or tackle more of the system first?

**Comment by tech4GT** [comment]
@jywarren We should wait on this one I guess, I want to work on this a little more before merging this is. Thanks :)

**Comment by tech4GT** [comment]
closing this in favor of #308

**Pull request closed without merging**